streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```diff
@@ -87,47 +88,113 @@ public void addStatistics(final String storeName,
+            thread.setName(storeName.replace(" ", "-") + "-RocksDB-metrics-recorder");
+            thread.start();
+            recordingAlreadyStarted = true;
         }
     }
 
-    public void record() {
-        // TODO: this block of record calls merely avoids compiler warnings.
-        //       The actual computations of the metrics will be implemented in the next PR
-        bytesWrittenToDatabaseSensor.record(0);
-        bytesReadToDatabaseSensor.record(0);
-        memtableBytesFlushedSensor.record(0);
-        memtableHitRatioSensor.record(0);
-        memtableAvgFlushTimeSensor.record(0);
-        memtableMinFlushTimeSensor.record(0);
-        memtableMaxFlushTimeSensor.record(0);
-        writeStallDurationSensor.record(0);
-        blockCacheDataHitRatioSensor.record(0);
-        blockCacheIndexHitRatioSensor.record(0);
-        blockCacheFilterHitRatioSensor.record(0);
-        bytesReadDuringCompactionSensor.record(0);
-        bytesWrittenDuringCompactionSensor.record(0);
-        compactionTimeAvgSensor.record(0);
-        compactionTimeMinSensor.record(0);
-        compactionTimeMaxSensor.record(0);
-        numberOfOpenFilesSensor.record(0);
-        numberOfFileErrorsSensor.record(0);
+    private void recordLoop() {
+        log.info("Started with recording interval {}", recordingInterval.toString());
+        while (!stopRecording) {
+            recordOnce();
+            try {
+                Thread.sleep(recordingInterval.toMillis());
+            } catch (final InterruptedException exception) {
+                // do nothing and wait until next iteration
+            }
+        }
+        log.info("Stopped", storeName);
+    }
+
+    public void recordOnce() {
+        long bytesWrittenToDatabase = 0;
+        long bytesReadFromDatabase = 0;
+        long memtableBytesFlushed = 0;
+        long memtableHits = 0;
+        long memtableMisses = 0;
+        long blockCacheDataHits = 0;
+        long blockCacheDataMisses = 0;
+        long blockCacheIndexHits = 0;
+        long blockCacheIndexMisses = 0;
+        long blockCacheFilterHits = 0;
+        long blockCacheFilterMisses = 0;
+        long writeStallDuration = 0;
+        long bytesWrittenDuringCompaction = 0;
+        long bytesReadDuringCompaction = 0;
+        long numberOfOpenFiles = 0;
+        long numberOfFileErrors = 0;
+        synchronized (statisticsToRecord) {
+            for (final Statistics statistics : statisticsToRecord.values()) {
+                bytesWrittenToDatabase += statistics.getTickerCount(TickerType.BYTES_WRITTEN);
+                bytesReadFromDatabase += statistics.getTickerCount(TickerType.BYTES_READ);
+                memtableBytesFlushed += statistics.getTickerCount(TickerType.FLUSH_WRITE_BYTES);
+                memtableHits += statistics.getAndResetTickerCount(TickerType.MEMTABLE_HIT);
+                memtableMisses += statistics.getAndResetTickerCount(TickerType.MEMTABLE_MISS);
```

Comment:
At the moment, hit ratios consists only of the current value measured in the recording interval. Should we change this to AVG, MIN, and MAX?